app/models/course/material/folder.rb

```diff
@@ -15,6 +15,29 @@ class Course::Material::Folder < ActiveRecord::Base
   validate :validate_name_is_unique_among_materials
   validates_with FilenameValidator
 
+  # @!attribute [r] material_count
+  #   Returns the number of files in current folder.
+  calculated :material_count, (lambda do
+    Course::Material.select { count('*') }.
+      where { course_materials.folder_id == course_material_folders.id }
+  end)
+
+  # @!attribute [r] children_count
+  #   Returns the number of subfolders in current folder.
+  calculated :children_count, (lambda do
+    select('COUNT(*)').
```

Comment:
`SELECT  "course_material_folders".*, (SELECT count('*') FROM "course_material_folders" INNER JOIN "course_material_folders" "children_course_material_folders" ON "children_course_material_folders"."parent_id" = "course_material_folders"."id") AS children_count FROM "course_material_folders" WHERE "course_material_folders"."id" = $1 LIMIT 1  [["id", 197]]`

This is a example query, it selects `course_material_folders` so that the number is for all folders. Yah, I was wondering if it can be simplified by squeel